.github/workflows/publish_docker_images.yml

```diff
@@ -55,12 +57,131 @@ jobs:
-          tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+          outputs: type=image,name=${{ env.REGISTRY_IMAGE }},push-by-digest=true,name-canonical=true,push=true
+
+      - name: Export digest
+        run: |
+          mkdir -p /tmp/digests
+          digest="${{ steps.build.outputs.digest }}"
+          touch "/tmp/digests/${digest#sha256:}"
+
+      - name: Upload digest
+        uses: actions/upload-artifact@v3
+        with:
+          name: digests
+          path: /tmp/digests/*
+          if-no-files-found: error
+          retention-days: 1
+
+  docker-arm:
+    runs-on: buildjet-4vcpu-ubuntu-2204-arm
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            ${{ env.REGISTRY_IMAGE }}
+          tags: |
+            type=edge,branch=main
+            type=edge,branch=main,suffix=-bullseye
+            type=semver,pattern={{version}}
+            type=semver,pattern={{version}},suffix=-bullseye
+            type=ref,event=tag
+          labels: |
+            org.opencontainers.image.title=Quickwit
+            maintainer=Quickwit, Inc. <hello@quickwit.io>
+            org.opencontainers.image.vendor=Quickwit, Inc.
+            org.opencontainers.image.licenses=AGPL-3.0
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_ACCESS_TOKEN }}
+
+      - name: Retrieve commit date, hash, and tags
+        run: |
+          echo "QW_COMMIT_DATE=$(TZ=UTC0 git log -1 --format=%cd --date=format-local:%Y-%m-%dT%H:%M:%SZ)" >> $GITHUB_ENV
+          echo "QW_COMMIT_HASH=$(git rev-parse HEAD)" >> $GITHUB_ENV
+          echo "QW_COMMIT_TAGS=$(git tag --points-at HEAD | tr '\n' ',')" >> $GITHUB_ENV
+
+      - name: Build and push
+        uses: docker/build-push-action@v4
+        id: build
+        with:
+          context: .
+          platforms: linux/arm64
+          build-args: |
+            QW_COMMIT_DATE=${{ env.QW_COMMIT_DATE }}
+            QW_COMMIT_HASH=${{ env.QW_COMMIT_HASH }}
+            QW_COMMIT_TAGS=${{ env.QW_COMMIT_TAGS }}
+          labels: ${{ steps.meta.outputs.labels }}
+          outputs: type=image,name=${{ env.REGISTRY_IMAGE }},push-by-digest=true,name-canonical=true,push=true
+
+      - name: Export digest
+        run: |
+          mkdir -p /tmp/digests
+          digest="${{ steps.build.outputs.digest }}"
+          touch "/tmp/digests/${digest#sha256:}"
+
+      - name: Upload digest
+        uses: actions/upload-artifact@v3
+        with:
+          name: digests
+          path: /tmp/digests/*
+          if-no-files-found: error
+          retention-days: 1
+
+  merge:
```

Comment:
What does this step do? It seems to create and inspect a new manifest but I don't see a push.

Reply:
Yeah, the documentation is unclear. The only hint about pushing is the `dry-run` option: `--dry-run |   | Show final image instead of pushing`. Thanks for the explanation.